### PR TITLE
Update UI to decouple DetailCommand and UiManager and reflect immediate UI changes

### LINKED ISSUE #175: Changes by commands that modify contact's information should be reflected immediately in PersonViewPanel

### PR DESCRIPTION
**Pull request by imkwokyong**
Tinkered with the UI and found a way to decouple DetailCommand and UiManager, and to reflect updates immediately on the PersonViewPanel. Generalized it to a few other commands that modify the details of the contact.

The AddressBook has an attribute called detailedPerson, which is initialized to null.
In separate Commands, the person(toEdit, toSchedule, toMark etc) will be used to set the detailed person through model -> modelManager -> AddressBook.
Then in MainWindow#executeCommand, after every command is executed and a CommandResult is returned, in the finally block the PersonViewPanel will be updated with the latest detailedPerson, which is received through logic -> logicManager -> model -> modelManager -> AddressBook.
The implementation is similar to filteredPersons, but I didn't make a JavaFX implementation (with Observeable and listeners) because of time limitations and JavaFX skill issue :( **( I love JavaFX :D** )

What's to be expected:
1. When user starts the app, the user should expect to see this PersonViewPanel (it doesn't matter if the address book is populated, because detailedPerson is initialized as null)
image
2. Any command that modifies the contact, like EditCommand, ScheduleCommand etc, will cause the detailedPerson in AddressBook to be updated and reflected immediately in PersonViewPanel. (It doesn't matter if preceding PersonViewPanel is not about the contact that is edited)
- Before edit
image
- After edit
image

3. DeleteCommand and ClearCommand will cause the PersonViewPanel to be cleared, reverting to the one in 1.
- Something could be done for DeleteCommand (like the Panel remains the same if deleted contact is not the one in the Panel, but for now I can't think of a way to do it)
5. Note that the PersonViewPanel will be restored to the default in 1. after exiting and relaunching the app.

Those are the tentative edits that I make in this branch. Making this PR a draft because it's not complete yet (pending resolve merge conflicts and open for suggestions)
This PR can be discarded if there is a better implementation.

Closes #175

### CODE REVIEW

File: src/main/java/connexion/ui/PersonViewPanel.java [hunk 1]
```diff
@@ -58,24 +58,30 @@ public class PersonViewPanel extends UiPart<Region> {
...
-        schedule.setText(person.getSchedule()
-                .map(Schedule::getDetailString).orElse(""));
-        scheduleName.setText(person.getScheduleName()
-                .map(Object::toString).orElse("No scheduled meetings with this person yet"));
-        //note.setText(person.getNote.getDetailString());
-        note.setText("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam volutpat, quam ac vehicula.");
+        if (person == null) {
+            this.person = null;
+            this.noteHeader.setText("");
+            this.name.setText("Nobody is here :(");
```
Comment (Angelyxx) [inline src/main/java/connexion/ui/PersonViewPanel.java:64]: remember to change this to `this.name.setText("Welcome to Connexion!");` :)

File: src/main/java/connexion/ui/PersonViewPanel.java [hunk 2]
```diff
@@ -58,24 +58,30 @@ public class PersonViewPanel extends UiPart<Region> {
...
+            schedule.setText(person.getSchedule()
+                    .map(Schedule::getDetailString).orElse(""));
+            scheduleName.setText(person.getScheduleName()
+                    .map(Object::toString).orElse("No scheduled meetings with this person yet"));
+            //note.setText(person.getNote.getDetailString());
+            note.setText("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam volutpat, quam ac vehicula.");
+        }
     }
 
     /**
```
Comment (Angelyxx) [inline src/main/java/connexion/ui/PersonViewPanel.java:87]: Remember to remove this PersonViewPanel() constructor since it's not used anymore!

**Review by Angelyxx — COMMENTED** [review]
Tinkered around and all LGTM besides changing the display line in PersonViewPanel for the null value + removing the unused constructor!

**Comment by imkwokyong** [comment]
Moved welcome message to ResultDisplay, so that's the first thing they see.

File: src/main/java/connexion/ui/PersonViewPanel.java (hunk 1 above)
Comment (imkwokyong) [reply src/main/java/connexion/ui/PersonViewPanel.java]: New changes to the UI, see below

**Review by xavierpok — APPROVED** [review]
Tests look sufficient except a few missed lines, & UI was not covered by tests previously. LGTM.

**Review by Angelyxx — APPROVED** [review]
LGTM!

**Pull request merged**